### ISSUE TITLE
Configure error for the test SMS_Ln1.ne4_ne4.FC5AV1C-L.summit_ibm.eam-chem_pp

### ISSUE THREAD

**Issue by minxu74** [E3SM-Project/E3SM#4499]
There is an error during the configuration of  the test SMS_Ln1.ne4_ne4.FC5AV1C-L.summit_ibm.eam-chem_pp  on Summit using the IBM XL compiler.

```
      ERROR: Command: '/autofs/nccs-svm1_home1/mxu/MyGit/e3sm_RHEL8_summit/components/eam/bld/configure -s -ccsm_seq -ice none -ocn docn -caseroot /gpfs/alpine/cli115/proj-shared/mxu/e3sm_
scratch/SMS_Ln1.ne4_ne4.FC5AV1C-L.summit_ibm.eam-chem_pp.20210827_084737_ptyayy -comp_intf mct  -spmd -spmd -smp -nosmp -dyn se -dyn_target theta-l -res ne4np4   -mach summit -phys default -
clubb_sgs -microphys mg2 -chem linoz_mam4_resus_mom_soag -rain_evap_to_coarse_aero -nlev 72 -bc_dep_to_snow_updates -usr_mech_infile /autofs/nccs-svm1_home1/mxu/MyGit/e3sm_RHEL8_summit/cime/
../components/eam/chem_proc/inputs/pp_linoz_mam4_resus_mom_soag.in ' failed with error '*** Chem preprocessor FAILED.
         See: /gpfs/alpine/cli115/proj-shared/mxu/e3sm_scratch/SMS_Ln1.ne4_ne4.FC5AV1C-L.summit_ibm.eam-chem_pp.20210827_084737_ptyayy/Buildconf/eamconf/chem_proc/chem_mech.doc' from dir '/g
pfs/alpine/cli115/proj-shared/mxu/e3sm_scratch/SMS_Ln1.ne4_ne4.FC5AV1C-L.summit_ibm.eam-chem_pp.20210827_084737_ptyayy/Buildconf/eamconf'

```
There is no chem_mech.doc file in the directory mentioned in the above error message.

**Comment by singhbalwinder** [comment]
Hi @minxu74 : Would you please check to see if there exists a file MAKE.out or something similar in the case directory (Buildconf/eamconf/chem_proc/)? That file generally has the error info. We should also change this misleading error message and point users to MAKE.out file instead if it exists.

**Comment by minxu74** [comment]
@singhbalwinder Thanks. Yes. There is a file named MAKE.out in the directory, but there is no any error information in it. Instead, there is a core file in the Buildconf/eamconf. 

BTW: The MAKE.out is at the /gpfs/alpine/cli115/proj-shared/mxu/e3sm_scratch/SMS_Ln1.ne4_ne4.FC5AV1C-L.summit_ibm.eam-chem_pp.20210827_084737_ptyayy/Buildconf/eamconf/chem_proc/MAKE.out.

**Comment by singhbalwinder** [comment]
If you got a core file, it looks like the executable compiled fine but encountered a runtime error. Most likely we need to debug it to find out what is going on. Simplest thing to do would be to change the optimizations to zero and see if it works. To do that, please modify `components/eam/chem_proc/src/Makefile` for the xlf compiler section (around line 154ish)and change the optimization from O2/O3 to O0 (or O1).

Also, if you have "campp" executable in your source code (at `components/eam/chem_proc/`), please delete it and run the test again as a new test (by deleting the old one or with a new tag). Sometimes if a code is copied from somewhere else (some other machine/architecture), the old executable also gets copied which will obviously won't work on a different machine/architecture.

**Comment by minxu74** [comment]
@singhbalwinder Thanks. Yes. There is a "campp" file in the source code. After it was removed, the configure error was gone.  When I traced the error, I found that campp was generated when I ran the tests for the gnu compiler. After the tests for gnu compiler were finished, I ran the test for the ibm compiler and then the gnu-compiled campp caused the error. So I guess that the campp file should not stay in the source code tree and need to move to the bld directory.

**Comment by singhbalwinder** [comment]
Glad that it is working fine now. The system uses pre-built exe if it exists. May be the GNU compiled exe didn't find the needed libraries (as ibm env was loaded), so it failed. If you think the issue is resolved, feel free to close this issue.

**Comment by rljacob** [comment]
Does the compiled "campp" file always appear in the source directory?  I'm surprised this doesn't happen more often on any machine with 2 compilers that people switch between.

**Comment by minxu74** [comment]
The compiled "campp" does not exist in the fresh cloned source directory. But once a case is compiled, the compiled "campp" will appear in the source directory. If users switch to another compiler, the build system won't recompile it when it is found in the source directory. I guess that the gnu-compiled and xl-compiled binaries are more different and use different libraries on Summit compared to other machines. So the problem only appeared on Summit.

**Comment by rljacob** [comment]
That's probably it.  @jgfouca see above.  There's a build artifact in the EAM source that needs to be removed with doing "case.build --clean".

**Comment by singhbalwinder** [comment]
There is one more point to note: `campp` is only created/used if we invoke chemistry preprocessor. Chemistry preprocessor is invoked in just one test: `SMS_Ln1.ne4_ne4.FC5AV1C-L.*.eam-chem_pp`. For all other tests, we use pre-existing F90 chemistry files present in the folders starting with `pp_` in directory `components/eam/src/chemistry`.

**Comment by minxu74** [comment]
@singhbalwinder @rljacob Thanks a lot for help.

**Comment by rljacob** [comment]
@minxu74 so you weren't switching compilers in a single casedir?  You were running tests twice with different compilers on the same SRCDIR?   I think the solution then is either:  the campp file has to be moved to the EXEROOT and somehow called from there.  Or it gets renamed to "campp.compiler" so gnu will look for/build campp.gnu and not use campp.ibm.

**Comment by minxu74** [comment]
@rljacob Yes. you are right. I was not switching compilers in a single casedir.  I just ran tests twice using different compilers in the same source directory as follows:
```
./create_test e3sm_integration --compiler gnu
```
After the above tests were finished, I ran the same command in the same src directory, but used the different compiler
```
./create_test e3sm_integration --compiler ibm
```
Since the campp file is a compiled executive file, similar to other temporary files that are generated during compilation, I am for the solution to move it to the EXEROOT, or the build directory. Thanks.
